### PR TITLE
Do not run assets on release branch push and fix version of EVE-OS in EdenGCP

### PR DESCRIPTION
**Pull request by giggsoff**
We prepare assets only for default branch (snapshot) and for tagged version. So we should ignore release branches itself, only tags on them.
Now we can see errors in assets workflow (Error response from daemon: manifest for lfedge/eve:8.5-kvm-amd64 not found: manifest unknown: manifest unknown). Let's remove that trigger and wait for the new tag on patch branch before
publishing.

EdenGCP workflow uses workflow_run trigger. In that caseactions/checkout will pull default branch which is not expected. Let's
get version of EVE-OS from the source tree.

### CODE REVIEW

**Comment by eriknordmark** [comment]
Looks like you removed all pushing for release branches. Is that the case? Or does the release branch push happen somewhere else?

**Comment by giggsoff** [comment]
> Looks like you removed all pushing for release branches. Is that the case? Or does the release branch push happen somewhere else?

We will not push non-tagged versions from release (patch) branches. I put comment about that "Publishing of patch branch updates ends with odd tags on dockerhub (for
example lfedge/eve:8.5.1-1-g344965dda) and lead to error in assets
workflow (Error response from daemon: manifest for lfedge/eve:8
.5-kvm-amd64 not found: manifest unknown: manifest unknown).". If such docker tags expected, we should adjust Assets workflow.

**Comment by giggsoff** [comment]
I reworked the PR to not prepare assets on release branch update. We still will make them on tag push event on release branch, it is separate trigger.

File: .github/workflows/eden_gcp.yml [hunk 1]
```diff
@@ -78,19 +78,20 @@ jobs:
...
+          echo "TAG=$(make -C eve --no-print-directory version)" >> "$GITHUB_ENV"
+      # Use the latest version of EDEN with subset of tests from EVE-OS repo
       - name: Prepare eden
         run: |
-          if [ "${{ matrix.backend }}" == "rol" ] || ! [ -f ${{ github.workspace }}/eve/tests/eden/eden-version ]; then
-             EDEN_VERSION=lfedge/eden:0.7.0
-          else
-             EDEN_VERSION=$(cat ${{ github.workspace }}/eve/tests/eden/eden-version)
-          fi
+          EDEN_VERSION=lfedge/eden:0.7.0
```
Comment (eriknordmark) [inline .github/workflows/eden_gcp.yml:94]: Comment about latest and then using 0.7.0 seems to be at odds with each other.

Do we not want to pick up eden-version from the eve repo?

**Review by eriknordmark — APPROVED** [review]
Kick of tests.

**Review by rvs — COMMENTED** [review]
LGTM

**Pull request merged**